### PR TITLE
Attempt to remove component from render world if not extracted.

### PR DESCRIPTION
**Pull request by tychedelia**
# Objective

Ensure that components that are conditionally extracted do not linger in the render world when not extracted from the main world.

## Solution

If the `ExtractComponent` returns `None`, we'll remove the render world component. I think this is the most sensible behavior here. In the future if there really is a use case for keeping the previous render component around, we could add a `Option<Self::Out>` parameter for the previous render component to the method, or something similar. I think that this follows the principle of least surprise here relative to what `None` would suggest and the way that render nodes are typically written. The alternative would be to add an `enabled` field to pretty much every camera settings component, or duplicate the extraction condition as #15856 does.

## Testing

`transmission` no longer crashes.

## Migration Guide

Components that implement `ExtractComponent` and return `None` will cause the extracted component to be removed from the render world.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
I like this fix! Simple and relatively robust.

**Comment by alice-i-cecile** [comment]
Does this fully fix #15871?

**Comment by tychedelia** [comment]
> Does this fully fix #15871?

Only the ones that implement `ExtractComponent` (which is most of them), theres a few that are still manual extraction systems we need to audit.

File: crates/bevy_render/src/extract_component.rs [hunk 1]
```diff
@@ -205,6 +205,8 @@ fn extract_components<C: ExtractComponent>(
     for (entity, query_item) in &query {
         if let Some(component) = C::extract_component(query_item) {
             values.push((entity, component));
+        } else {
+            commands.entity(entity).remove::<C>();
```
Comment (JMS55) [inline crates/bevy_render/src/extract_component.rs:209]: @alice-i-cecile What is the behavior of remove() when the entity lacks the component already? No-op, or panic?

The docs don't say: https://dev-docs.bevyengine.org/bevy_ecs/system/commands/struct.EntityCommands.html#method.remove

File: crates/bevy_render/src/extract_component.rs [hunk 2]
```diff
@@ -222,6 +224,8 @@ fn extract_visible_components<C: ExtractComponent>(
         if view_visibility.get() {
             if let Some(component) = C::extract_component(query_item) {
                 values.push((entity, component));
+            } else {
+                commands.entity(entity).remove::<C>();
             }
         }
```
[the remark this answers is not in the document]
Comment (tychedelia) [reply crates/bevy_render/src/extract_component.rs:230]: Hmmm, good question. I think if the extracted component isn't in visible entities it should never be rendered but we should probably do this to be consistent.

**Review by IceSentry — COMMENTED** [review]
I think this behaviour should be documented on the `ExtractComponent` trait

File: crates/bevy_render/src/extract_component.rs (hunk 1 above)
Comment (alice-i-cecile) [reply crates/bevy_render/src/extract_component.rs:209]: No-op.

File: crates/bevy_render/src/extract_component.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (JMS55) [reply crates/bevy_render/src/extract_component.rs:230]: Do we not extract invisible entities? 

Not sure how I feel about removal in this case. I'd rather leave it for now I think.

**Review by BenjaminBrienen — APPROVED** [review]

**Review by JMS55 — APPROVED** [review]

File: crates/bevy_render/src/extract_component.rs [hunk 3]
```diff
@@ -205,6 +206,8 @@ fn extract_components<C: ExtractComponent>(
     for (entity, query_item) in &query {
         if let Some(component) = C::extract_component(query_item) {
             values.push((entity, component));
+        } else {
+            commands.entity(entity).remove::<C>();
```
[the remark this answers is not in the document]
Comment (tychedelia) [reply crates/bevy_render/src/extract_component.rs:210]: Gosh duh. In many cases they're the same but good catch.

**Pull request merged**